Mvp 95

minSsan opened the pull request:
# 🔍 What is this PR for? | PR 설명

> PR 을 만들게 된 원인 / 문제상황 / 해결법 등을 설명해주세요.

- [Jira MVP-95 링크](https://caregiver-frontend.atlassian.net/browse/MVP-95)

# 📝 Changes | 핵심 변경사항

> 리뷰어가 알아야 할 핵심 변경사항이 있다면 미리 말해주세요.

- 

# 📷 Screenshots | 스크린샷

> PR 을 설명할 스크린샷을 첨부할 수 있다면, 첨부해주세요.

-

# 🤦🏻‍♂️ Help me | 도와주세요

> 해결하지 못한 점이 있어, 도움이 필요한 경우 남겨주세요.

- 예약 상세정보 스크린을 확인할 수 없어서, 작업 적용 결과를 확인하지 못한 상황
    - 기존 레이아웃 때문에 원하는대로 결과가 안 나왔을 것으로 추정🥺

smnchoi committed:
Update pull_request_template.md

hojkim77 committed:
MVP-76: (진행중) 리뷰 조회 api 작성 및 연결

hojkim77 committed:
MVP-76: 리뷰 작성 api 작성 및 연결 - 데이터 페칭 문제로인해 model 사용 필요

minSsan committed:
MVP-95 build: popover view 라이브러리 추가

minSsan committed:
MVP-95 feat: 클라이언트 예약내역 상세페이지 스크린 - popover 컴포넌트 추가

minSsan committed:
build: yarn lock file

minSsan committed:
build: ios project file

hojkim77 committed:
MVP-76: 리뷰 조회 api 작성 및 연결 완료

hojkim77 committed:
MVP-76: 리뷰 정렬 기능 추가

hojkim77 committed:
MVP-73: (진행중)보호자 입장 예약 모델 작성

smnchoi committed:
Merge pull request #203 from Care-Giver/release/0.0.3

⚡️Release/0.0.3 [Sync to develop]

hojkim77 committed:
feat: 예약 및 예약Store 모델 작성

smnchoi committed:
Create License

smnchoi committed:
yarn add react-native-config

smnchoi committed:
extra step for Android

https://github.com/lugg/react-native-config?tab=readme-ov-file#extra-step-for-android

smnchoi committed:
typescript config

https://github.com/lugg/react-native-config?tab=readme-ov-file#typescript-declaration-for-your-env-file

smnchoi committed:
fix: "react-native-dotenv" 를 사용했던 코드,  "react-native-config" 를 사용하도록 수정

smnchoi committed:
remove "react-native-dotenv"

smnchoi committed:
Merge pull request #207 from Care-Giver/package/react-native-config

⚡️Package/react native config

minSsan committed:
feat: js, ts 파일 내 key 값 .env 파일로 옮김

minSsan committed:
feat: strings.xml  키값 이동, google-services.json 파일 gitignore 추가

smnchoi committed:
fix(babel.config.js): react-native-dotenv 관련 코드 삭제

smnchoi committed:
Merge pull request #208 from Care-Giver/MVP-103

Mvp 103

hojkim77 committed:
fix: PR리뷰 수정사항 반영

smnchoi committed:
Merge pull request #209 from Care-Giver/develop

⚡️Sync develop to release/1.0.0

smnchoi committed:
Merge branch 'release/1.0.0' into MVP-95

smnchoi committed:
MVP-95 fix: from prop 내 요소를 View 로 감싸줘야 제대로 popover 가 표출 됨

smnchoi commented on app/screens/_CLIENT/booking-stack/booking-detail-screen/booking-detail-screen.tsx:
```diff
@@ -194,10 +195,12 @@ export const BookingDetailScreen: FC<
                       onRequestClose={() => setShowPopover(false)}
                       from={
-                        <>
+                        <View style={{ flexDirection: "row", justifyContent: "space-between" }}>
```
@minSsan 
`from` prop 내 요소를 View 로 감싸줘야 제대로 popover 가 표출 됩니다.

[React.Fragment `<>...</>`](https://react.dev/reference/react/Fragment) 는 리액트에서는 큰 문제가 되지 않지만,

RN 에서는 지양하는 것이 좋을 것 같습니다. RN 에서는, View 를 대신 사용해주세요.
현재 사용중인 `react-native-popover-view` 라이브러리에서 처럼, React.Fragment 를 사용하면 문제가 되는 경우가 있습니다.
이번 문제의 원인은, 추정컨데, React.Fragment 를 사용하면 from 영역 계산이 제대로 이루어지지 않아 발생하는 문제 같습니다.

참고:
- `react-native-popover-view` 패키지 소스코드, 영역 계산 함수: 
- https://github.com/SteffeyDev/react-native-popover-view/blob/master/src/Geometry.ts#L72

hojkim77 committed:
fix: 수정 확인 버튼 활성화 범위 확장

smnchoi committed:
fix(package.json): yarn start 명령어에 --reset-cache 옵션 추가

smnchoi committed:
fix(결제수단 UI): 카카오페이를 제외하고 모두 숨김

hojkim77 committed:
fix: 펫시터 서비스 등록 스크린 카드 UI 깨짐 문제 해결

smnchoi committed:
feat: preRegister 함수 추가

hojkim77 committed:
fix: PR리뷰 수정사항 반영_2

smnchoi committed:
feat(카카오페이): 결제 요청 로직 구현

smnchoi committed:
feat(카카오페이): 결제 결과 로직 구현

smnchoi committed:
chore(make-booking-screen): import 정리

smnchoi committed:
Merge pull request #212 from Care-Giver/MVP-108

⚡️Mvp 108: 카카오페이 결제 구현

smnchoi committed:
Upgrade version to "1.0.0-alpha.1"

smnchoi committed:
fix(iOS): 배포시, "수출 규정 준수 정보" 무시하도록 수정

https://develop-ssooo.tistory.com/88

smnchoi committed:
Merge pull request #210 from Care-Giver/MVP-105

Mvp 105

smnchoi committed:
chore: 위탁 스크린 주석 수정

smnchoi committed:
Merge pull request #211 from Care-Giver/MVP-107

MVP 107

smnchoi committed:
[후기 작성 기능 추가, UT 관련 더미데이터 삭제] Merge branch 'release/1.0.0' into MVP-76

# Conflicts:
#	app/components/review-box/review-box.tsx
#	app/models/index.ts
#	app/screens/_CLIENT/booking-stack/all-bookings-screen/all-bookings-screen.tsx
#	app/screens/_CLIENT/caregiver-detail-stack/all-reviews-screen/all-reviews-screen.tsx
#	app/screens/_CLIENT/caregiver-detail-stack/caregiver-detail-information-screen/caregiver-detail-information-screen.tsx

smnchoi committed:
fix: setSelectedComment state 로직 개선

smnchoi committed:
fix(댓글 수정): commentId 제대로 설정하도록 수정 - 기존에는 commentId 에 작성자 id 가 잘못 할당되어 있었음

smnchoi committed:
fix: __commentator__ 에 optional chain 추가, updateComment 에러메시지 수정

smnchoi committed:
fix(댓글 작성/수정): debounce 추가, disabled 추가

smnchoi committed:
fix(댓글 스크린 헤더): 작성 or 수정 헤더 텍스트 수정

smnchoi committed:
fix(주로 사용중인 버튼 컴포넌트들): debounce 추가

smnchoi committed:
Merge pull request #204 from Care-Giver/MVP-76

Mvp 76

smnchoi committed:
fix(cg-calendar): 비어있는 일정은 "none" 으로 수정

smnchoi committed:
fix(펫시터): 수락한 예약, "케어 일정" 기준으로 오름차순 정렬

smnchoi committed:
fix(booking-list): 좌우 패딩 추가

smnchoi committed:
refactor: booking-list -> cg-booking-list 리네임, booking-info-card 이전

smnchoi committed:
feat(cg-manage-booking-screen): "월"별 예약 필터링 기능 추가

smnchoi committed:
fix(cg-booking-detail-screen): 케어 일정 ERR 표출 오류 수정

smnchoi committed:
fix(care-summary): petIds -> pets prop 으로 수정 - 펫시터 모드일 때도 사용하기 위해 수정 함

smnchoi committed:
feat(펫시터-상세-스크린): 댓글이 없을 때 텍스트 추가

smnchoi committed:
Merge branch 'release/1.0.0.alpha.2' into MVP-95

# Conflicts:
#	app/screens/_CLIENT/booking-stack/booking-detail-screen/booking-detail-screen.tsx

smnchoi committed:
refactor: past-booking -> booking-info-card 리네임

smnchoi committed:
feat(booking-info-card): "신청한 예약"과 "지난 예약" 케이스에서 모두 사용 가능하도록 개선

smnchoi committed:
feat(booking-detail-screen): "전화하기", "신고하기" UI 숨김

smnchoi committed:
fix(all-bookings-screen): startTime 기준으로 정렬 - "확정된 예약", "신청된 예약"

smnchoi committed:
refactor: cancel-reservation-screen 삭제 후, cancel-booking-bottom-sheet-modal 으로 모듈화

smnchoi committed:
feat(보호자): "신청한 예약" [예약 취소] 기능 추가

smnchoi committed:
feat: 어댑터 패턴 적용, booking-info-card-adaptor

smnchoi committed:
fix(package.json): yarn ios 실행시 "--reset-cache" 옵션 추가
